app/controllers/api/tests_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Api::TestsController < Api::BaseController
+  before_action :doorkeeper_authorize!, only: %i[update create]
+  def update
+    begin
+      params.require(:id)
+      params.require(:name)
+      params.require(:status)
+      raise ActionController::BadRequest.new("Wrong format") unless params[:id].is_a? Integer
```

Comment:
The `Api::TestsController` is using a manual way to require parameters and validate them. Consider using ActiveModel Serializers or similar mechanisms for parameter validation to make the code cleaner and more maintainable.